src/dbnode/storage/shard.go

```diff
@@ -1787,14 +1804,84 @@ func (s *dbShard) Bootstrap(
 	// (will be passed to series.Bootstrap() as BlockState).
 	s.bootstrapFlushStates()
 
+	multiErr := xerrors.NewMultiError()
+	bootstrapResult, err := s.loadSeries(bootstrappedSeries, true)
+	if err != nil {
+		multiErr = multiErr.Add(err)
+	}
+	s.emitBootstrapResult(bootstrapResult)
+
+	// From this point onwards, all newly created series that aren't in
+	// the existing map should be considered bootstrapped because they
+	// have no data within the retention period.
+	s.Lock()
+	s.newSeriesBootstrapped = true
+	s.Unlock()
+
+	// Find the series with no data within the retention period but has
+	// buffered data points since server start. Any new series added
+	// after this will be marked as bootstrapped.
+	s.forEachShardEntry(func(entry *lookup.Entry) bool {
+		seriesEntry := entry.Series
+		if seriesEntry.IsBootstrapped() {
+			return true
+		}
+		_, err := seriesEntry.Bootstrap(nil, series.BootstrappedBlockStateSnapshot{})
+		multiErr = multiErr.Add(err)
+		return true
+	})
+
+	// Now that this shard has finished bootstrapping, attempt to cache all of its seekers. Cannot call
+	// this earlier as block lease verification will fail due to the shards not being bootstrapped
+	// (and as a result no leases can be verified since the flush state is not yet known).
+	if err := s.cacheShardIndices(); err != nil {
+		multiErr = multiErr.Add(err)
+	}
+
+	s.Lock()
+	s.bootstrapState = Bootstrapped
+	s.Unlock()
+
+	return multiErr.FinalError()
+}
+
+func (s *dbShard) Load(
+	series *result.Map,
+) error {
+	s.Lock()
+	// Don't allow loads until the shard is bootstrapped because the shard flush states need to be
+	// bootstrapped in order to safely load blocks. This also keeps things simpler to reason about.
+	if s.bootstrapState != Bootstrapped {
+		s.Unlock()
+		return errShardIsNotBootstrapped
+	}
+	s.Unlock()
+
+	_, err := s.loadSeries(series, false)
+	return err
+}
+
+func (s *dbShard) loadSeries(
+	series *result.Map,
+	bootstrap bool,
+) (dbShardBootstrapResult, error) {
+	if series == nil {
+		return dbShardBootstrapResult{}, nil
```

Comment:
Hm I'd be in preference of making this an error case, you ok with that?

Reply:
The reason it behaves this way is because its very common to call `series.Bootstrap(nil)` but I could move it so `Bootstrap(nil)` is fine and `Load(nil)` is an error

Reply:
Could we do that?  Make bootstrap(nil) valid but Load(nil) not valid?

Reply:
@robskillington yeah thats what I implemented